### PR TITLE
HTML parsing feature

### PR DESCRIPTION
**Pull request by ajayns**
This was getting a bit experimental, I tried multiple approaches with react-html-parser and react-serialize. This was the one that looked fine to me. 

Initially, I tried directly exporting by adding React to the plugin, but that was a bad decision. Now this approach currently handles all the links. 

Planning to iterate over this.

### CODE REVIEW

File: src/gatsby-node.js [hunk 1]
```diff
@@ -97,6 +130,12 @@ const processData = (data, baseUrl) => {
     }
   });
 
+  if (node.text) {
+    if (node.text['content-type'] === 'text/html') {
+      node.text.html = processHtml(node.text.data, baseUrl);
```
Comment (datakurre) [inline src/gatsby-node.js]: maybe node.text.react =

Comment (ajayns) [reply src/gatsby-node.js]: There's also node.text.jsx? What would be better here?

**Comment by datakurre** [comment]
jsx means jsx syntax; I assume react-serialize serialises “react component tree”, not jsx

**Comment by datakurre** [comment]
Exciting. I’ll try this before tomorrow.

**Comment by datakurre** [comment]
I got a console error
```
Error: link is a void element tag and must neither have `children` nor use `dangerouslySetInnerHTML`.
```
when rendering page with internal link.

**Comment by ajayns** [comment]
That is strange, I'll investigate and try to fix the issue. The thing was that react-parser-html when replacing tags preferably only takes in native HTML tags and not React components, so what I used was a workaround assuming it'll be handled right in the gatsby-site.

**Comment by datakurre** [comment]
@ajayns My last comment was not really issue in plugin, but issue that should be documented its users. The solution is to give proper component mapping for deserializer


```js
import { Link } from 'gatsby';

export default ({ data }) => (
    <div>
      {deserialize(data.text.react, {
        components: {
          Link: Link,
        },
      })}
   </div>
);
```

we could make that easier by shipping with "RichText"-component that maps gatsby link and image for the serializer.

**Comment by ajayns** [comment]
So for now in the tester, I'll ship it with the code you mentioned? And then later iterate when we have the RichText component?

**Comment by ajayns** [comment]
Also, is the fixtures updated with html containing an internal link? So that we can Travis to check if everything's working fine.

**Comment by datakurre** [comment]
@ajayns No, the fixture is not yet updated. You can do it:

```
make init-backend
```

always resets the backend

Then use Plone to create relative link between the pages.

```
make export-fixture
```

Updates the test fixture form Plone.

As an unfortunate side effect these commands do also update all UID-values in fixture (it's kind of proper behavior). You can either ignore all those changes when committing the fixture or commit them all.

**Comment by datakurre** [comment]
@ajayns A small real issue:

I assume it is our use of react-html-parser that bloats browser console with:

```
Warning: Each child in an array or iterator should have a unique "key" prop.
```

I believe this is only annoyance in our use case (we don't update rich text values once rendered in a way that would really require/benefit from defined keys), but I assume this could be prevented in react-html-parser by creating random unique key for each node.

**Comment by ajayns** [comment]
For testing: 
```
Scenario: Internal linking
  Go To  http://localhost:9000/testcases/a-page-with-internal-linking/
  Page should contain  Link to first-results
  Click link  Link to first-results
  Page should contain  First results
  Page should contain  The Google Summer of Code
```

Is that good enough?

**Comment by datakurre** [comment]
@ajayns You could also check that the links has correct attribute value as its href.

**Comment by datakurre** [comment]
I'll approve and merge this.

I'll figure out next steps for this during the weekend. Handling links to Plone images and files looks a bit more complicated. Also, we need to handle broken links somehow.

**Pull request merged**